### PR TITLE
DEPR: remove real, imag, put, and str.partition kwarg

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

**Comment by gfyoung** [comment]
These changes I thought would have been fine.

Seem to have instead triggered multiple test failures...

**Comment by jbrockmendel** [comment]
Looks like ptp behavior is fickle depending on numpy version, will revert that and save it for a later day

File: pandas/tests/test_strings.py [hunk 1]
```diff
@@ -2966,23 +2966,21 @@ def test_partition_with_name(self):
...
         # using sep -> no warning
         expected = values.str.partition(sep="_")
-        with tm.assert_produces_warning(FutureWarning):
-            result = values.str.partition(pat="_")
-            tm.assert_frame_equal(result, expected)
+        result = values.str.partition("_")
+        tm.assert_frame_equal(result, expected)
 
         # str.rpartition
         # using sep -> no warning
```
Comment (jorisvandenbossche) [inline pandas/tests/test_strings.py]: Small nit, but those comments are no longer up to date (also above)

**Comment by jreback** [comment]
thanks @jbrockmendel 

small followup indicated by @jorisvandenbossche  above

File: pandas/tests/test_strings.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/test_strings.py]: updated #29985 to remove these comments

**Pull request merged**